app/forms/firewall-rules-common.tsx

```diff
@@ -494,6 +494,12 @@ const ProtocolFilters = ({ control }: { control: Control<FirewallRuleValues> })
                   placeholder=""
                   validate={icmpCodeValidation}
                   transform={normalizeDashes}
+                  onKeyDown={(e) => {
+                    if (e.key === KEYS.enter) {
+                      e.preventDefault() // prevent full form submission
+                      submitProtocol(e)
+                    }
+                  }}
```

Comment:
This wasn't causing the flake because we weren't pressing enter in this field in the test, but it was wrong. You could submit the whole form by pressing enter in this field.